app/controllers/wikis_controller.rb

```diff
@@ -0,0 +1,63 @@
+class WikisController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+
+  def index
+    @wikis = Wiki.all
+  end
+
+  def show
+    @wiki = Wiki.find(params[:id])
+  end
+
+  def new
+    @wiki = Wiki.new
+  end
+
+  def create
+    @wiki = Wiki.new(wiki_params)
+    @wiki.user = current_user
+
+    if @wiki.save
+      redirect_to @wiki, notice: "Wiki was saved successfully."
+    else
+      flash.now[:alert] = "Error creating wiki. Please try again."
```

Comment:
This line is duplicated below.. You could refactor to extract this out, but not big enough to worry about it too much :)